src/main.rs

```diff
@@ -31,6 +33,42 @@ impl From<SupportedLanguage> for Language {
     }
 }
 
+enum OutputFile {
+    Stdout,
+    Disk {
+        staged: NamedTempFile,
+        output: OsString,
+    },
+}
+
+impl OutputFile {
+    fn new(path: Option<&str>) -> FormatterResult<Self> {
+        match path {
+            Some("-") | None => Ok(Self::Stdout),
+            Some(file) => Ok(Self::Disk {
+                staged: NamedTempFile::new()?,
+                output: file.into(),
+            }),
+        }
+    }
+
+    fn persist(self) -> FormatterResult<()> {
+        // NOTE Maybe it would be better to persist on Drop...or is that too magical?
```

Comment:
I don't think this is obvious Drop behaviour. What if you just decided you wanted to discard the output file entirely?